packages/opds-web-client/src/components/__tests__/OPDSCatalog-test.tsx

```diff
@@ -4,73 +4,52 @@ import { stub } from "sinon";
 import * as React from "react";
 import * as PropTypes from "prop-types";
 import { shallow } from "enzyme";
+import { ReactReduxContext } from "react-redux";
 
 import OPDSCatalog from "../OPDSCatalog";
 import Root, { RootProps } from "../Root";
-import buildStore from "../../store";
-import { mockRouterContext } from "./routing";
+import { mockRouterContext } from "../../__mocks__/routing";
 
 describe("OPDSCatalog", () => {
   let props = {
     collectionUrl: "collection url",
     bookUrl: "book url",
     proxyUrl: "proxy url",
     navigate: stub(),
-    pathFor: (collectionUrl: string, bookUrl: string): string => { return "path"; },
+    pathFor: (collectionUrl: string, bookUrl: string): string => {
+      return "path";
+    },
     bookData: {
       id: "book id",
       title: "book title",
       url: "book url"
     },
     pageTitleTemplate: (c, b) => "test title",
-    epubReaderUrlTemplate: (a) => "test reader url"
+    epubReaderUrlTemplate: a => "test reader url"
   };
   let context = mockRouterContext();
 
-  it("creates a store for Root if not given one", () => {
-    let wrapper = shallow(
-      <OPDSCatalog {...props} />,
-      { context,
-        childContextTypes: {
-          router: PropTypes.object,
-          pathFor: PropTypes.func
-        }
-      }
-    );
-    let root = wrapper.find<RootProps>(Root);
-    expect(root.props().store).to.be.ok;
-  });
-
-  it("passes state to Root if given one", () => {
-    let store = buildStore();
-    let state = store.getState();
-    let wrapper = shallow(
-      <OPDSCatalog {...props} initialState={state} />,
-      { context,
-        childContextTypes: {
-          router: PropTypes.object,
-          pathFor: PropTypes.func
-        }
-      }
-    );
-    let root = wrapper.find<RootProps>(Root);
-    expect(root.props().store.getState()).to.deep.equal(state);
-  });
-
   it("passes props to Root", () => {
-    let wrapper = shallow(
-      <OPDSCatalog {...props} />,
-      { context,
-        childContextTypes: {
-          router: PropTypes.object,
-          pathFor: PropTypes.func
-        }
+    let wrapper = shallow(<OPDSCatalog {...props} />, {
+      context,
+      childContextTypes: {
+        router: PropTypes.object,
+        pathFor: PropTypes.func
       }
-    );
-    let root = wrapper.find<RootProps>(Root);
+    });
+    /**
+     * This is painfully fragile, but must be done to make
+     * enzyme render the child function beneath ReactReduxContext.Consumer
+     */
+    const root = wrapper
+      .dive()
+      .dive()
+      .dive()
+      .dive();
```

Comment:
Is this because it needs to render `Root` which is under `ReactReduxContext` which is under `OPDSStore`?  That's three levels or am I counting the dives incorrectly?

Reply:
Yea the dives are difficult to count, but essentially enzyme doesn't have great support for child functions and we need it to render that in order to test it properly. We alternatively could `mount()` the whole thing, but that would render all the way down the tree which would also not be a great way to test, so I thought this was the best bad option